inference-test-tool/run.py

```diff
@@ -23,46 +23,52 @@
 import pydicom
 import numpy as np
 import test_inference_mask, test_inference_boxes, test_inference_classification
-from utils import create_folder
+from utils import create_folder, DICOM_BINARY_TYPES
 
 
 from utils import load_image_data, sort_images
 
-SEGMENTATION_MODEL = "SEGMENTATION_MODEL"
-BOUNDING_BOX = "BOUNDING_BOX"
-CLASSIFICATION_MODEL = "CLASSIFICATION_MODEL"
-OTHER = "OTHER"
 
-def upload_study_me(file_path, model_type, host, port, output_folder, attachments, override_inference_command=None, send_study_size=False, include_label_plots=False, route='/'):
+def save_secondary_captures(json_response, output_folder_path, multipart_data):
+    secondary_capture_parts = [
+        p for p in json_response['parts'] if p['binary_type'] in
+        ['dicom_structured_report', 'dicom_secondary_capture']
+    ]
+
+    # Create DICOM files for secondary capture outputs
+    for index in range(len(secondary_capture_parts)):
+        file_path = os.path.join(
+            output_folder_path, 'sc_{}.dcm'.format(index)
+        )
+        with open(file_path, 'wb') as outfile:
+            outfile.write(multipart_data.parts[index + 1].content)
+
+def upload_study_me(file_path,
+                    host,
+                    port,
+                    output_folder,
+                    attachments,
+                    override_inference_command='',
+                    send_study_size=False,
+                    include_label_plots=False,
+                    route='/',
+                    request_study_path='',
+                    request_options={}):
     file_dict = []
     headers = {'Content-Type': 'multipart/related; '}
 
     images = load_image_data(file_path)
     images = sort_images(images)
 
-    if model_type == BOUNDING_BOX:
-        print("Performing bounding box prediction")
-        inference_command = 'get-bounding-box-2d'
-    elif model_type == SEGMENTATION_MODEL:
-        if images[0].position is None:
-            # No spatial information available. Perform 2D segmentation
-            print("Performing 2D mask segmentation")
-            inference_command = 'get-probability-mask-2D'
-        else:
-            print("Performing 3D mask segmentation")
-            inference_command = 'get-probability-mask-3D'
```

Comment:
just realized we might be losing some functionality here, are the callers of this meant to be passing in `-c get-probability-mask-3D` now?

Reply:
That would be a reasonable assumption. Basically the send-inference-request tool was trying to be smart sometimes, but not other times, in determining the command name.

We don't have to worry about this being a breaking chance, since the individual models' CI builds are locked down to their specific version of inference SDK rather than always using 'master', so there's no chance of this breaking the existing CI builds. This tool is also not used in production in any way, just a test tool for rapid internal testing of models.